### DISCUSSION TITLE
Route search params reloads state on link hover

### DISCUSSION THREAD

**Discussion by BenJeau** [TanStack/router#1078] [Q&A]
Hi,

I've looked around and this seems to be possibly a bug (unless I have misconfigured how I'm using search params within TanStack router), but when I hover a link, even when I didn't click on the link, it loads the new search param within the state of my component.

Here's an example of it happening, you can see in the devtool (and URL, but forgot to grab it in the recording) that it didn't update when I hovered yet the code updated within the name field from **Micor** to **Testing**.

![Screen Recording 2024-01-24 at 8 35 46 PM](https://github.com/TanStack/router/assets/22248828/7aeee8df-7b2a-4661-b55e-7f17b98dc332)

I use the router with TanStack Router where I have `defaultPreload: "intent"` set within my main router, but still wouldn't expect the behaviour of search params affecting on hover.

To add on, I get the search params within the relevant page component using `Route.useSearch();`

**Comment by BenJeau** [comment]
I was able to reproduce it here https://stackblitz.com/edit/github-3yrcd5 the main culprit seems to be because of `defaultPreloadStaleTime: 0`

**Reply by BenJeau** [reply]
I was following the docs here https://tanstack.com/router/v1/docs/guide/data-loading#passing-all-loader-events-to-an-external-cache saying to use it for external loaders such as TanStack Query, but it seems to have other side effects?

**Reply by SeanCassiere** [reply]
Could you try adding this in your route?
```tsx
export const Route = createFileRoute({
  // ...,
  loaderDeps: ({ search }) => ({ search }), // <- this right here
});
```

**Reply by schiller-manuel** [reply]
@SeanCassiere  yes this makes the problem go away: https://stackblitz.com/edit/github-3yrcd5-1o2qe2?file=src%2Fmain.tsx
but is this the expected behavior without loaderDeps?

**Reply by SeanCassiere** [reply]
Probably isn't.

It's something I remember from this [issue](https://github.com/TanStack/router/issues/749) and eventually loaderDeps came out.

We could prevent committing the location to primary if in `preload` I think.

**Reply by BenJeau** [reply]
I've tried that and it seems to have gone away, but I've just updated to the latest version and switch my router from `new Router` to `createRouter` and everything seems fixed without having to do the `loaderDeps: ({ search }) => ({ search })` trick
